storage/factory/common.go

```diff
@@ -90,7 +97,17 @@ func FindLastEpochFromStorage(
 		epochDirs = append(epochDirs, file.Name())
 	}
 
-	return getLastEpochFromDirNames(epochDirs)
+	lastEpoch, err := getLastEpochFromDirNames(epochDirs)
+	if err != nil {
+		return 0, err
+	}
+
+	if lastEpoch == 0 {
```

Comment:
what happens in a setting with storage pruning on and the node was shut down in epoch 0?
Why not injecting the setting that tells you if storage pruning is on or off instead of making assumptions?

Reply:
You can't tell this because the config may have changed since last run. So you don't know if the last time the node was online it had storage pruning off. If the node shuts down and goes back online in epoch 0, the code will be executed. Anyway, also in this PR I have to return also the last round, so each time you'll have to open the persister